pkg/k8s/node.go

```diff
@@ -60,10 +61,19 @@ func ParseNode(k8sNode *slim_corev1.Node, source source.Source) *nodeTypes.Node
 		if ip == nil {
 			scopedLog.WithFields(logrus.Fields{
 				logfields.IPAddr: addr.Address,
-				"type":           addr.Type,
+				logfields.Type:   addr.Type,
 			}).Warn("Ignoring invalid node IP")
 			continue
 		}
+		_, typeFound := addressTypes[addr.Type]
+		if typeFound && len(k8sNode.Spec.PodCIDRs) != 2 {
+			scopedLog.WithFields(logrus.Fields{
+				logfields.Node: k8sNode.Name,
+				logfields.Type: addr.Type,
+			}).Warn("Detected multiple IPs of the same address type, Cilium will only consider the first IP in the Node resource")
+		} else {
+			addressTypes[addr.Type] = struct{}{}
+		}
```

Comment:
In order to account for dual-stack environments, we should define a new type that will be the key of the `addressTypes` map. It will look something like

```go
type foo struct {
  typ slim_corev1.NodeAddressType
  int ipFamily
}
```

Then the map will allow us to handle same IP family and address type at the same time.